packages/suite/src/views/firmware/index.tsx

```diff
@@ -194,12 +211,13 @@ const Firmware = ({ closeModalApp, resetReducer, firmware, device }: Props) => {
             useFixedHeight
             data-test="@firmware/index"
             heading={Component.Heading}
-            bottomBar={<Buttons>{Component.BottomBar}</Buttons>}
+            bottomBar={!modal && <Buttons>{Component.BottomBar}</Buttons>}
             totalProgressBarSteps={stepsInProgressBar.length}
             currentProgressBarStep={getCurrentStepIndex()}
             hiddenProgressBar={false}
         >
-            {Component.Body}
+            {modal && <InnerModalWrapper>{modal}</InnerModalWrapper>}
+            {!modal && Component.Body}
```

Comment:
Would be cleaner with an if-else statement.

Reply:
I changed that but I dont like the proposed solution. two lines if negated if statement feels more readable to me. probably depends on individual taste